jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/annotations/AnnotationDiagnosticsCollector.java

```diff
@@ -220,11 +235,10 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
                             }
 
                             if (Flags.isStatic(method.getFlags())) {
-                                Diagnostic diagnostic = new Diagnostic(
-                                        methodRange,
-                                        "A method with the annotation @PreDestroy must not be static.");
-                                diagnostic.setCode(AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_STATIC);
-                                completeDiagnostic(diagnostic);
+                            	String msg = "A method with the annotation @PreDestroy must not be static.";
+                            	Diagnostic diagnostic=createDiagnostic(method, unit, msg,
+                            			AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_STATIC);
+                            	diagnostic.setData(method.getElementType());
```

Comment:
Indentation is off.
To keep things consistent and avoid duplicate work, either update all of the corresponding diagnostics to make use of the `createDiagnostic` method, or change this diagnostic to:

```java
String msg = "A method with the annotation @PreDestroy must not be static.";
Diagnostic diagnostic = new Diagnostic(methodRange, msg);
diagnostic.setCode(AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_STATIC);
completeDiagnostic(diagnostic);
diagnostic.setData(method.getElementType());
diagnostics.add(diagnostic);
```

Reply:
solved